src/sqlancer/stonedb/gen/StoneDBTableDeleteGenerator.java

```diff
@@ -37,24 +39,25 @@ public SQLQueryAdapter getQuery() {
         }
         sb.append(" FROM ");
         sb.append(randomTable.getName());
-        if (Randomly.getBoolean()) {
-            sb.append(" AS ");
-            sb.append(globalState.getSchema().getRandomTable().getName());
-        }
         if (Randomly.getBoolean()) {
             sb.append(" WHERE ");
             sb.append(StoneDBToStringVisitor.asString(new StoneDBExpressionGenerator(globalState)
                     .setColumns(randomTable.getColumns()).generateExpression()));
         }
         if (Randomly.getBoolean()) {
             sb.append(" ORDER BY ");
-            sb.append(Randomly.fromOptions(
-                    randomTable.getColumns().stream().map(AbstractTableColumn::getName).collect(Collectors.toList())));
+            sb.append(String.join(", ", Randomly.fromOptions(
+                    randomTable.getColumns().stream().map(AbstractTableColumn::getName).collect(Collectors.toList())))
+                    .replace('[', '(').replace(']', ')'));
         }
         if (Randomly.getBoolean()) {
             sb.append(" LIMIT ");
-            sb.append(new Randomly().getInteger());
+            sb.append(r.getInteger(0, (int) randomTable.getNrRows(globalState)));
         }
+        addExpectedErrors();
         return new SQLQueryAdapter(sb.toString(), errors);
     }
+
+    private void addExpectedErrors() {
```

Comment:
As mentioned before, I don't think we need to introduce such empty methods.

Reply:
@ZhengLin-Li Can you remove them?

Reply:
done